lib/client/db/dbcmd/dbcmd.go

```diff
@@ -465,6 +465,11 @@ func (c *CLICommandBuilder) getRedisCommand() *exec.Cmd {
 		if c.options.caPath != "" {
 			args = append(args, []string{"--cacert", c.options.caPath}...)
 		}
+
+		// Set SNI when sending to remote web proxy.
+		if c.options.localProxyHost == "" {
+			args = append(args, []string{"--sni", c.tc.WebProxyHost()}...)
```

Comment:
I'm struggle to grasp why this is needed 
Does in case of non local proxy the --sni = c.tc.WebProxyHost() should be always part of the `-h` destination  host address  so the connection string will look like  `--sni teleport.example.com -h teleoprt.exampe.com:443` ?

Reply:
Yes. Without `--sni teleport.example.com`, I get the following error on the server side
```
2022-09-15T14:54:15Z WARN [ALPN:PROX] Failed to handle client connection. error:[
ERROR REPORT:
Original Error: *errors.errorString acme/autocert: missing server name
Stack Trace:
/go/teleport/lib/srv/alpnproxy/proxy.go:389 github.com/gravitational/teleport/lib/srv/alpnproxy.(*Proxy).handleConn
/go/teleport/lib/srv/alpnproxy/proxy.go:327 github.com/gravitational/teleport/lib/srv/alpnproxy.(*Proxy).Serve.func1
/usr/local/go/src/runtime/asm_amd64.s:1571 runtime.goexit
```
I have acme on and i think this is part of the default web TLS config. Alternatively, we can try to allow this on the server side. WDYT?

Reply:
Ah I see. I think that teleport cloud require proper SNI to route the request to correct tenant so it should be fixed in the client side.